### DISCUSSION TITLE
Add tracking for browser javascript errors

### DISCUSSION THREAD

**Discussion by christopherpickering** [umami-software/umami#1082] [Ideas]
This tool is great and covers the area of privacy analytics... do you think it can be slightly expanded to also monitor browser errors coming through the javascript console? There are a few tools doing this (rollbar, trackjs etc) but none of them are privacy conscious. 

There is a nice script out there call [jsnlog ](https://www.jsnlog.com) also on [npm](https://www.npmjs.com/package/jsnlog), that can be used to catch the errors and send them back to the web server.

What do you think about adding another checkbox on the website config, somewhere here: 
![image](https://user-images.githubusercontent.com/17788706/160404475-75d9e9a1-ae2d-43ea-afbb-0d9b3331432e.png)

To also track javascript console message? The output is simple and can be added to one additional db table. From a privacy standpoint, there it nothing needing recorded except the user agent and error messages.

It would be an awesome feature to have. I cannot find any open source tools (with a web ui like umami) that do this job.

**Comment by mikecao** [comment]
There will be more expanded features in the future and this is one of them.
